.github/workflows/actions/check-git-context/action.yml

```
@@ -0,0 +1,11 @@
+name: 'Check Git Context'
+description: 'checks for a dirty git context, failing if the context is dirty'
+runs:
+  using: composite
+  steps:
+    - name: Git status check
+      # here we check that there are no changed / new files.
+      # we use `git status`, grep out the build zip used throughout CI,
+      # and check if there are more than 0 lines in the output.
```

maybe we should just update this comment real quick to say

```suggestion
      # here we check that there are no changed / new files.
      # we use `git status`, grep out the build zip used throughout CI,
      # and check if there are more than 0 lines in the output.
      # if there are more than 0 lines (which indicates new files written
      # during the CI run) we print out the return value of `git status`
```

or something along those lines, just to accurately describe what we do here

---

👍 I dig it - done in https://github.com/ionic-team/stencil/pull/3625/commits/21d0c85b55a2549b408aae182af2808fdcea1a65